### PR TITLE
Update FAQ - provide some "pro" for DNSSEC

### PR DESCRIPTION
**Pull request by thecapacity**
I understand the point that HTTPS provides the "all three" but the document semi-reads as though DNSSEC has *no* value, which I don''t believe to be the intention. 

Specifically, I think DNSSEC has positives such as:
* Combined with HTTPS and HTTP/2 as a "defense in depth"
* Provides "one of the legs" of support for Authenticity that benefits not HTTP* protocols.

Just didn't want the document to read like "DNSSEC isn't worth doing..."

### CODE REVIEW

File: pages/faq.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ However, DNS resolution is just one aspect of securely communicating on the inte
 
 * No major web browsers inform the user when DNSSEC validation fails, limiting its strength and enforceability.
 
-HTTPS guarantees the confidentiality and integrity of communication between client and server, and web browsers have rigorous and evolving HTTPS enforcement policies.
+HTTPS guarantees the confidentiality and integrity of communication between client and server, and web browsers have rigorous and evolving HTTPS enforcement policies. However, DNSSEC is an important component of providing assurance of *authenticity*, particularly for traffic that does not utilize HTTP, HTTPS, and HTTP2 protocols.
```
Comment (alex) [inline pages/faq.md]: This isn't really true. DNSSEC, insofar as it does anything, only provides for authenticity of the DNS records, which says nothing about about the contents of the traffic. Indeed any MITM attacker able to send false DNS records is equally able to send false traffic (that is not authenticated by TLS or similar)

**Comment by konklone** [comment]
Yeah, @thecapacity, these two DNSSEC FAQ answers are meant to address questions raised about DNSSEC as it relates to M-15-13. Comments and information about whether DNSSEC is in and of itself worth doing are not something we want to get into here. 

FWIW, you may find it helpful that the formal policy itself includes a paragraph about DNSSEC that explicitly states that the HTTPS policy doesn't rescind or conflict with the DNSSEC policy:

https://https.cio.gov/#challenges-and-considerations

So I'm a :-1: on merging, but will leave this open for a bit in case there's more comment.

**Comment by thecapacity** [comment]
Totally see your point - I wasn't aware that this was an M-15-13 specific page, and in general would suggest shying away from memo specific pages so we can start weaving the implications on Federal IT. 

As you say, since M-15-13 does include text then perhaps that gives us a guiding "star", the memo states: "Once DNS resolution is complete, DNSSEC does not ensure the privacy or integrity of communication between a client and the destination IP. HTTPS provides
this additional security. "

So maybe the text could be something to that effect, e.g. "

"HTTPS guarantees the confidentiality and integrity of communication between client and server, and web browsers have rigorous and evolving HTTPS enforcement policies. Note this compliments, continues, and extends previous Federal efforts such as M-08-23 to secure network communications."

**Comment by thecapacity** [comment]
for @alex I agree the wording isn't quite right - but I think it's important to point out that HTTPS will _not_ provide any of the aforementioned benefits for other ports and protocols (unless of course they're tunneled over HTTPS).

So at least DNSSEC provides one leg of the stool in terms of records validation.

My comment wouldn't be that DNSSEC is sufficient, I just wouldn't want an agency to read this and think they didn't need DNSSEC, because it does provide some value / defense in depth - particularly since not all the web is HTTP.

$.02 - I'm not an expert on the technical matters, just some experience in the ways people use english to justify avoiding some policies about implementing them..

**Comment by konklone** [comment]
> My comment wouldn't be that DNSSEC is sufficient, I just wouldn't want an agency to read this and think they didn't need DNSSEC, because it does provide some value / defense in depth - particularly since not all the web is HTTP.

So, this is getting into my **personal** feelings -- and not OMB's, GSA's, or 18F's -- but I don't see it quite the same way. HTTPS and DNSSEC aren't comparable "defenses in depth", because HTTPS+HSTS can hard-fail and DNSSEC cannot. 

Browsers don't enforce DNSSEC today, though some networks enforce DNSSEC for .gov specifically. When a domain has HSTS enabled (especially if it's preloaded), it can fail closed, and an attacker that manipulates DNS will be required to produce a certificate (since they won't be able to downgrade the request to plain HTTP). It is possible to use DNS validation to get a certificate, but this will require attacking DNS between a site and its CA, rather than a site and its visitor.

The only extent to which DNSSEC is an additional layer of defense in depth is that it gives some DNSSEC-aware networks and clients (which don't include browsers) a sort of "best effort" confidence that it managed to resolve the IP correctly. If the site is HTTP (or lacks HSTS) then an active attacker has full control over the connection, regardless of DNSSEC. If the site is HTTPS+HSTS and is being attacked by someone with the ability to get in the middle of DNS records, perhaps a client can use it to get some extra confidence about what to do (but no browser exposes this for user review). 

Basically, a site owner that supports DNSSEC has no universal guarantees about what clients will do under attack conditions. There are no reporting mechanisms, and enforcement is sporadic and currently only at the network level.

I'm not arguing in this comment that the government should alter its policy on DNSSEC. I **am** saying that I _personally_ don't see M-15-13 as complementing the DNSSEC policy. It's not a case of "HTTPS takes care of this part of it, and DNSSEC takes care of this other part of it". Their threat models and guarantees are radically different, and I'm more concerned about risking someone taking away from the text a simpler (and incorrect) impression that "HTTPS does HTTP, and DNSSEC does DNS".

**Comment by thecapacity** [comment]
Fair point(s) @konklone - I was primarily trying to point out DNSSEC's applicability towards non-HTTP(s) protocols, i.e. agencies are more than just websites and HTTP(S)-based APIs.

**Comment by konklone** [comment]
> agencies are more than just websites and HTTP(S)-based APIs.

They definitely are, and I'm definitely not trying to ignore that. I'm going to close the issue as mentioned above, but this has generated some helpful discussion and I'm probably going to reference it going forward.
